SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/Utils/ApplicationLifecycleHelper.cs

```diff
@@ -46,17 +45,23 @@ public ApplicationLifecycleHelper()
                 // the start
                 _started = true;
             }
-            try
+
+            // Subscribe to unhandled errors events
+            CoreApplication.UnhandledErrorDetected += (sender, eventArgs) =>
             {
-                Application.Current.UnhandledException += (sender, eventArgs) =>
+                try
                 {
-                    UnhandledExceptionOccurred?.Invoke(sender, new UnhandledExceptionOccurredEventArgs(eventArgs.Exception));
-                };
-            }
-            catch (Exception exception)
-            {
-                MobileCenterLog.Warn(MobileCenterLog.LogTag, "Cannot subscribe to unhandled exceptions from XAML application", exception);
-            }
+                    // intentionally propagating exception to get the exception object that crashed the app.
+                    eventArgs.UnhandledError.Propagate();
+                }
+                catch (Exception exception)
+                {
+                    UnhandledExceptionOccurred?.Invoke(sender, new UnhandledExceptionOccurredEventArgs(exception));
+
+                    // If we don't throw exception - app will not be crashed. We need to throw to not change the app behavior.
+                    throw;
```

Comment:
According to the [docs](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.core.coreapplication):

> As part of an app's life cycle management code, developers can inspect error data (UnhandledError ) from an UnhandledErrorDetected event, and choose whether to mark the error as handled. If the error is marked handled in the event data, then execution can continue. If the error is not marked handled, the app and its process will be terminated.

So why re-throw? Also, do we want to raise the event if it has already been marked handled? Is that possible?